src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -110,10 +118,58 @@ case class Cluster(clusterName: ClusterName,
   def projectNameString: String = s"${googleProject.string}/${clusterName.string}"
 }
 
+object MachineConfig {
+  implicit val machineConfigSemigroup = new Semigroup[MachineConfig] {
+    def combine(config1: MachineConfig, config2: MachineConfig): MachineConfig = {
```

Comment:
definitely rename this to `fallback` or something if you can...

Reply:
As discussed with faces: there's a philosophical question of which way around the arguments to `|+|` should be. There are two options:

1. We take the config on the left, and if any of its values are `None`, overwrite those values with the config on the right
2. We take the config on the left, and overwrite _all_ of its values from the config on the right, unless the value on the right is `None`

You've implemented one of these and my question is: which is the more intuitive definition? Perhaps you could ask experienced users of semigroups what they think the behaviour should be.